### PR TITLE
Include admin sets in collection search builders.

### LINKED ISSUE #1587: Collections: Modify My Collections search builder to return all editable collections and admin sets
### Descriptive summary

My Collections list of collections should include all collections and admin sets for which the current user has edit_access.

### Rationale

Per UX redesign, My Collections will include all editable collections and admin sets.

### Expected behavior

The list of collections should include all collections and admin set for which the current user has edit_access.

### Actual behavior

The list of collections includes the collections the current user deposited.

#### Current Implementation
Search builder specified:  [app/controllers/hyrax/my/collections_controller.rb](https://github.com/samvera/hyrax/blob/collections-sprint/app/controllers/hyrax/my/collections_controller.rb#L4-L6)

Current Search Builder:  [Hyrax::My::CollectionsSearchBuilder](https://github.com/samvera/hyrax/blob/collections-sprint/app/search_builders/hyrax/my/collections_search_builder.rb)

which inherits from: [Hyrax::My::SearchBuilder](https://github.com/samvera/hyrax/blob/collections-sprint/app/search_builders/hyrax/my/search_builder.rb)

#### Proposed Implementation

Rewrite the search builder such that each of the following are included:  (pseudocode)

```
solr_parameters[:fq] << "edit_access_group_ssim:#{current_user.groups}"
solr_parameters[:fq] << "edit_access_person_ssim:#{current_user}"
solr_parameters[:fq] << "has_model_ssim:AdminSet"
solr_parameters[:fq] << "has_model_ssim:Collection"
```

The challenging bit is that the `edit_access_person_ssim` for AdminSets does not include the person who created the admin set.  Collections do put the depositor in the `edit_access_person_ssim`. 

And the thing that makes this even more challenging is that the AdminSet keeps the id of the person who created the AdminSet in the `creator_ssim`.  And the Collection uses `creator_tesim` for the conceptual creator of the collection.  This is not tied to a user in the system and does not grant the person identified in `creator_tesim` any access to the Collection.

`has_model_ssim:AdminSet` uses field `creator_ssim:user@email.com`
`has_model_ssim:Collection` uses field `creator_tesim:John Smith`

The question is, can solr docs other than Admin Sets have a `creator_ssim` field?  Is it safe to assume that if we find `creator_ssim:user@email.com` AND our current user is `user@email.com` then this is a positive match?

Question:   How to get solr to do an OR on a field like has_model_ssim?

### Steps to reproduce the behavior

1. Create a few admin sets with various group and user participants
1. Create a few collections with various group and user participants
1. Login as a user that has edit access to multiple collections and admin sets
1. Dashboard -> Repository Contents -> Collections

Verify that all collections and admin sets for which the current user has edit_access are included in the list.

### Related work

#1509 Collections: Update UI/UX of the listing of collections in the All/My Collections index

### PR DESCRIPTION
**Pull request by jonathandixon**
Fixes #1587

This PR updates the search builders to include admin sets for the "all collections" and "my collections" dashboard pages.

Changes proposed in this pull request:
* Update the my collection search builder to include all collections and admin sets the current user has manage access for.
* Create a new search builder for the "All Collections" tab to include admin sets.
* Update the corresponding presenters and views to support Admin sets.

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by elrayle** [comment]
@jonathandixon I did testing with this and it behaved as expected.  Which is a bit unexpected.  The code that updates the solr-doc with correct edit and read access is not merged.  I haven't fully looked through the code to see why it worked anyway.  I would expect the search builder to try and answer all questions from solr unless it absolutely can't.  Does it make sense to you that the following scenario should work now without the solr doc getting proper access update?

* non-admin user A 
    * creates two collections (e.g. Col1 and Col2)  
    * shares Col1 with user B by adding as manager
* non-admin user B logs in
    * goes to Dashboard -> Collections
    * sees Col1, but not Col2

Looking in solr, Col1 does NOT have edit access for user B.  Looking in permissions table, user B is a manager.

FYI, I have most of the code done that will fix the edit access updates in the solr doc.  I just need to add more testing.

**Comment by elrayle** [comment]
Per discussion at standup, it looks like the search builder is querying the permission tables for manage access.  This is how the admin set search builder works.  I have a question in slack dev to ask why admin set search builder doesn't just use the solr doc edit access.

File: config/locales/hyrax.en.yml [hunk 1]
```diff
@@ -574,11 +576,14 @@ en:
         action:
           add_collection_to_collection: "Add collection to another collection"
           add_to_collection:        "Add to Collection"
+          admin_set_confirmation:  "Deleting this administrative set from is permanent. Click OK to delete this administrative set, or Cancel to cancel this operation"
```
Comment (elrayle) [inline config/locales/hyrax.en.yml]: Typo in... "Deleting this administrative set from is permanent."  
Should be?...  ""Deleting this administrative set from **the repository** is permanent."

File: app/views/hyrax/my/_collection_action_menu.html.erb [hunk 2]
```diff
@@ -14,27 +14,29 @@
...
+        <%= link_to hyrax.dashboard_collection_path(id),
+              class: 'itemicon itemtrash delete-collection-button',
+              title: t("hyrax.dashboard.my.action.delete_collection") do %>
+          <%= t("hyrax.dashboard.my.action.delete_collection") %>
+        <% end %>
+      </li>
+      <li role="menuitem" tabindex="-1">
+        <%= link_to '#',
+          class: 'itemicon add-collection-to-collection',
+          title: t("hyrax.dashboard.my.action.add_collection_to_collection"),
```
Comment (elrayle) [inline app/views/hyrax/my/_collection_action_menu.html.erb]: Depositors and managers can add collection to collection.

File: config/locales/hyrax.en.yml (hunk 1 above)
Comment (elrayle) [reply config/locales/hyrax.en.yml]: There should be a check before adding the delete action.  The modal for delete should either show the above message with the option to delete or cancel OR it should show a message "The administrative set is not empty.  Remove all items before deleting."

Checking on actual wording of the can't delete modal.

Comment (elrayle) [reply config/locales/hyrax.en.yml]: I think you can leave the message for now.  The delete action and modal messages have a separate issue #1507.  I would just fix the typo so it makes sense as it is right now.

File: app/views/hyrax/my/_collection_action_menu.html.erb [hunk 3]
```diff
@@ -14,20 +14,23 @@
...
+      </li>
+      <li role="menuitem" tabindex="-1">
+        <%= link_to hyrax.dashboard_collection_path(id),
+              class: 'itemicon itemtrash delete-collection-button',
+              title: t("hyrax.dashboard.my.action.delete_collection") do %>
+          <%= t("hyrax.dashboard.my.action.delete_collection") %>
+        <% end %>
+      </li>
+    <% end %>
+    <%# TODO: Only managers and depositors can add collections to collections. %>
```
Comment (jonathandixon) [inline app/views/hyrax/my/_collection_action_menu.html.erb]: I added this as a TODO for now. It looks like `Hyrax::Collections::PermissionService` has a method stubbed that can handle this.

**Pull request merged**